cloudius-systems/capstan
# Capstan:Migrate to Go Modules and Update APIs

As a Go project, Capstan uses godep as its dependency manager,
however, as of Go 1.11, Go's new dependency management system(Go
Modules) is introduced officially, more and more Go projects use Go
Modules to make dependency version information explicit and easier to
manage, this patch makes Capstan migrate to Go Modules and update
codes to use the latest APIs, here are details:

- Delete Godeps directory, add go.mod and go.sum file, which are
  generated by `go mod` command and declare Capstan project is managed
  with Go Modules now
- Update packages import path, change urfave/cli from v1 to v2, change
  cheggaaa/pb from v0 to v3, all packages are newest released version
- Update code to adapt new APIs according to the old/new API doc
- Update the installation.md, use Go 1.11+ to build/manage Capstan on
  Module Mode

See [Go Modules Blogs](https://blog.golang.org/using-go-modules) to learn more

Signed-off-by: kayhaw <xiaokedou123456@gmail.com>
